src/mlpack/core/tree/cover_tree/cover_tree_impl.hpp

```diff
@@ -1745,29 +1745,36 @@ void CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>::serialize(
     parent = NULL;
   }
 
-  ar & BOOST_SERIALIZATION_NVP(dataset);
-  ar & BOOST_SERIALIZATION_NVP(point);
-  ar & BOOST_SERIALIZATION_NVP(scale);
-  ar & BOOST_SERIALIZATION_NVP(base);
-  ar & BOOST_SERIALIZATION_NVP(stat);
-  ar & BOOST_SERIALIZATION_NVP(numDescendants);
-
   bool hasParent = (parent != NULL);
-  ar & BOOST_SERIALIZATION_NVP(hasParent);
-  ar & BOOST_SERIALIZATION_NVP(parentDistance);
-  ar & BOOST_SERIALIZATION_NVP(furthestDescendantDistance);
-  ar & BOOST_SERIALIZATION_NVP(metric);
+  ar & CEREAL_NVP(hasParent);
+  MatType* datasetTemp = const_cast<MatType*>(dataset);
+  if (!hasParent)
+    ar & CEREAL_POINTER(datasetTemp);
+
+  if (cereal::is_loading<Archive>())
+  {
+    dataset  = datasetTemp;
+  }
```

Comment:
I *think* you could avoid this by making `datasetTemp` a `MatType*&` and adjusting the `const_cast` accordingly.